features/bootstrap/FeatureContext.php

```diff
@@ -17,5 +41,120 @@ class FeatureContext implements Context
      */
     public function __construct()
     {
+        $this->methodResolver = new BehatMethodResolver();
+
+        $this->endpoint = (new FakeEndpointCreator())->create($this->methodResolver);
+    }
+
+    /**
+     * @When I send following payload:
+     */
+    public function whenISendTheFollowingPayload(PyStringNode $payload)
+    {
+        $this->lastResponse = $this->endpoint->index($payload->getRaw());
+    }
+
+    /**
+     * @Then last response should be a valid json-rpc result
+     */
+    public function thenLastResponseShouldHaveAResult()
+    {
+        $this->validateJsonRpcSuccessReponse(json_decode($this->lastResponse, true));
+    }
+
+    /**
+     * @Then last response should be a valid json-rpc error
+     */
+    public function thenLastResponseShouldHaveAnError()
+    {
+        $this->validateJsonRpcErrorReponse(json_decode($this->lastResponse, true));
```

Comment:
Handle json decoding in a method (+check json decode potential error) => easier for debugging